apollo-router/src/spec/query/change.rs

```diff
@@ -0,0 +1,611 @@
+                    self.hash_directive(&directive.node);
+                }
+            }
+            ExtendedType::Interface(i) => {
+                for directive in &i.directives {
+                    self.hash_directive(&directive.node);
+                }
+            }
+            ExtendedType::Union(u) => {
+                for directive in &u.directives {
+                    self.hash_directive(&directive.node);
+                }
+
+                for member in &u.members {
+                    self.hash_type_by_name(member.as_str());
+                }
+            }
+            ExtendedType::Enum(e) => {
+                for directive in &e.directives {
+                    self.hash_directive(&directive.node);
+                }
+
+                for (value, def) in &e.values {
+                    value.hash(self);
+                    for directive in &def.directives {
+                        self.hash_directive(directive);
+                    }
+                }
+            }
+            ExtendedType::InputObject(o) => {
+                for directive in &o.directives {
+                    self.hash_directive(&directive.node);
+                }
+
+                for (name, ty) in &o.fields {
+                    if ty.default_value.is_some() {
+                        name.hash(self);
+                        self.hash_input_value_definition(&ty.node);
+                    }
+                }
+            }
+        }
+    }
+
+    fn hash_type(&mut self, t: &ast::Type) {
+        match t {
+            schema::Type::Named(name) => self.hash_type_by_name(name.as_str()),
+            schema::Type::NonNullNamed(name) => {
+                "!".hash(self);
+                self.hash_type_by_name(name.as_str())
+            }
+            schema::Type::List(t) => {
+                "[]".hash(self);
+                self.hash_type(t);
+            }
+            schema::Type::NonNullList(t) => {
+                "[]!".hash(self);
+                self.hash_type(t);
+            }
+        }
+    }
+
+    fn hash_input_value_definition(&mut self, t: &Node<ast::InputValueDefinition>) {
+        self.hash_type(&t.ty);
+        for directive in &t.directives {
+            self.hash_directive(directive);
+        }
+        if let Some(value) = t.default_value.as_ref() {
+            self.hash_value(value);
+        }
+    }
+
+    fn hash_entities_operation(&mut self, node: &ast::OperationDefinition) -> Result<(), BoxError> {
+        use crate::spec::query::traverse::Visitor;
+
+        if node.selection_set.len() != 1 {
+            return Err("invalid number of selections for _entities query".into());
+        }
+
+        match node.selection_set.first() {
+            Some(Selection::Field(field)) => {
+                if field.name.as_str() != "_entities" {
+                    return Err("expected _entities field".into());
+                }
+
+                "_entities".hash(self);
```

Comment:
I think we already have a constant containing this value. If not, it could be interesting to have it and replace all existing occurences

Reply:
fixed in 5d8fc84c5

At some point we should have a central place for all those consts